src/app/book/add-edit-genre/add-edit-genre.component.ts

```diff
@@ -66,28 +67,14 @@ export class AddEditGenreComponent extends BaseComponent implements OnInit {
 
   public onAddGenre(): void {
     this.genre.name = this.form.get("name")?.value;
-    this.genreService.addGenre(this.genre)
-      .pipe(catchError(error => {
-        return throwError(error)
-      }),
-        takeUntil(this.destroy$))
-      .subscribe(
-        (response: any) => {
-          this.router.navigate(['books/genres'])
-        }
-      )
+    this.store.dispatch(addGenre(this.genre));
+    this.router.navigate(['books/genres'])
```

Comment:
`router.navigate` should be done when addGenre was done successfuly. 
One way is to dispatch a success action (ex `addGenreSuccess`) in genre effects and subscribe to it in component and then call this.router.navigate.
Second way is to call router.navigate directly from effect. Since we don't do any other operation calling router.navigate in effects is better.